src/Workspaces/MSBuildTest/NewlyCreatedProjectsFromDotNetNew.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.MSBuild.UnitTests
+{
+    public class NewlyCreatedProjectsFromDotNetNew : MSBuildWorkspaceTestBase
+    {
+        // The Maui templates require additional dotnet workloads to be installed.
+        // Running `dotnet workload restore` will install workloads but may require
+        // admin permissions. In addition a restart may be required after workload
+        // installation.
+        private const bool ExcludeMauiTemplates = true;
+
+        protected ITestOutputHelper TestOutputHelper { get; set; }
+
+        public NewlyCreatedProjectsFromDotNetNew(ITestOutputHelper output)
+        {
+            TestOutputHelper = output;
+        }
+
+        [ConditionalTheory(typeof(DotNetSdkMSBuildInstalled))]
+        [MemberData(nameof(GetCSharpProjectTemplateNames), DisableDiscoveryEnumeration = false)]
+        public async Task ValidateCSharpTemplateProjects(string templateName)
+        {
+            var ignoredDiagnostics = templateName switch
+            {
+                "blazor" or "blazorwasm" or "blazorwasm-empty" =>
+                    [
+                        "CS0246", // The type or namespace name {'csharp_blazor_project'|'App'} could not be found (are you missing a using directive or an assembly reference?)
```

Comment:
Is this a bug we should be tracking down at some point?

Reply:
Created bugs for the ignored diagnostics and added links to the comments.